src/main/java/org/reaktivity/nukleus/http2/internal/types/stream/Http2ErrorCode.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2016-2017 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.http2.internal.types.stream;
+
+import org.agrona.collections.Int2ObjectHashMap;
+
+public enum Http2ErrorCode
+{
+    NO_ERROR(0),
+    PROTOCOL_ERROR(1),
+    INTERNAL_ERROR(2),
+    FLOW_CONTROL_ERROR(3),
+    SETTINGS_TIMEOUT(4),
+    STREAM_CLOSED(5),
+    FRAME_SIZE_ERROR(6),
+    REFUSED_STREAM(7),
+    CANCEL(8),
+    COMPRESSION_ERROR(9),
+    CONNECT_ERROR(10),
+    ENHANCE_YOUR_CALM(11),
+    INADEQUATE_SECURITY(12),
+    HTTP_1_1_REQUIRED(13);
+
+    public final int errorCode;
+
+    private static class Codes
+    {
+        private static final Int2ObjectHashMap<Http2ErrorCode> CODES = new Int2ObjectHashMap<>();
+    }
+
+    Http2ErrorCode(int errorCode)
+    {
+        this.errorCode = errorCode;
+        Codes.CODES.put(errorCode, this);
+    }
+
+    static Http2ErrorCode from(long errorCode)
+    {
+        return Codes.CODES.get((int) errorCode);
```

Comment:
Replace with simple `switch` statement that will compile to a [`tableswitch`](https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-6.html#jvms-6.5.tableswitch) in JVM bytecode.

Removes need for `Codes` inner class.

Does parameter need to be a `long` rather than an `int`? If so, let's mask to `int` for `switch`, but first verify that high order byte is zero.